include/nbl/system/CWindowWin32.h

```diff
@@ -0,0 +1,136 @@
+#ifndef __C_WINDOW_WIN32_H_INCLUDED__
+#define __C_WINDOW_WIN32_H_INCLUDED__
+
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+
+#include "nbl/system/IWindow.h"
+
+namespace nbl {
+namespace system
+{
+
+class CWindowWin32 final : public IWindow
+{
+public:
+    using native_handle_t = HWND;
```

Comment:
need to hide the windows specific headers and definitions in a cpp file... yet still leave the `CWindowWin32::create` accesible to the user

Reply:
maybe `IWindowT` user accessible, and `CWindowT` with hidden OS specific header pollution, conditional compile and stuff?

Reply:
what would be in IWindowT? I don't see the point

Reply:
by `WindowT` I mean `WindowWin32`, `WindowX11`, `WindowWayland`, `WindowAndroid` etc

different windows differ by what they can offer, i.e. an android window is not really resizable, movable, etc.

Qt, and "foreign engine embedding" windows from extensions would not have much freedom either